From 322e1534917c53b3da22342cfa5898b2fa8e5af5 Mon Sep 17 00:00:00 2001
From: Tyler Veness <calcmogul@gmail.com>
Date: Wed, 20 Feb 2019 20:39:08 -0800
Subject: [PATCH] Don't remove unobservable or uncontrollable states from
 state-space model

---
 control/statesp.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/control/statesp.py b/control/statesp.py
index 9a7a69d..b970f3c 100644
--- a/control/statesp.py
+++ b/control/statesp.py
@@ -182,9 +182,6 @@ class StateSpace(LTI):
         if self.outputs != C.shape[0]:
             raise ValueError("C and D must have the same number of rows.")
 
-        # Check for states that don't do anything, and remove them.
-        self._remove_useless_states()
-
     def _remove_useless_states(self):
         """Check for states that don't do anything, and remove them.
 
-- 
2.21.0

